compiler/runtime/src/main/kotlin/runtime/driver.kt

```diff
@@ -0,0 +1,195 @@
+                    x + dims[index]
+                } else {
+                    x
+                }
+            }.toIntArray())
+        }
+    }
+}
+
+interface IWorldType {
+    fun getWorldView(world: World, dims: List<Int>, pos: List<Int>): IWorldView
+}
+
+@Deprecated("Use MultiWorldType")
+class WrappingWorldType: IWorldType {
+    override fun getWorldView(world: World, dims: List<Int>, pos: List<Int>): WrappingWorldView {
+        return WrappingWorldView(world, dims, pos)
+    }
+}
+
+@Deprecated("Use MultiWorldType")
+class EdgeWorldType(private val edge: Int): IWorldType {
+    override fun getWorldView(world: World, dims: List<Int>, pos: List<Int>): IWorldView {
+        return EdgeWorldView(world, dims, pos, edge)
+    }
+}
+
+class MultiWorldType(private val edge: Int, private val dimTypes: List<DimensionType>): IWorldType {
+    override fun getWorldView(world: World, dims: List<Int>, pos: List<Int>): IWorldView {
+        return MultiWorldView(world, dims, pos, edge, dimTypes)
+    }
+}
+
+open class Driver(private val worldConfig: WorldConfiguration, private val program: IProgram, private val worldType: IWorldType) {
+    var worldCurrent = World(worldConfig.dims)
+    var worldNext = World(worldConfig.dims)
+
+    open fun run() {
+        while(true) {
+            update()
+        }
+    }
+
+    open fun update() {
+        fun iterate(subDims: List<Int>, pos: List<Int> = listOf()) {
+            if (subDims.size == 1) {
+                var x = subDims[0] - 1
+                while (x >= 0) {
+                    val currentPos = pos + listOf(x)
+                    worldNext.setCell(
+                        *currentPos.toIntArray(),
+                        state = program.updateCell(
+                            worldView = worldType.getWorldView(worldCurrent, worldConfig.dims, currentPos)
+                        )
+                    )
+                    x--
+                }
+            } else {
+                var x = subDims[0] - 1
+                while (x >= 0) {
+                    @Suppress("UNCHECKED_CAST")
+                    iterate(subDims.subList(1, subDims.size), pos + listOf(x))
+                    x--
+                }
+            }
+        }
+
+        // Run iteration
+        iterate(worldConfig.dims)
+
+        // Flip worlds
+        val temp = worldCurrent
+        worldCurrent = worldNext
+        worldNext = temp
+    }
+}
+
+class World(private val dims: List<Int>) {
+    var world = generateWorld(dims)
+
+    private fun generateWorld(dims: List<Int>): MutableList<Any> {
+        var x = dims[0]
+        if (dims.size == 1) {
+            return generateSequence {
+                x--
+                if (x >= 0) {
+                    0
+                } else {
+                    null
+                }
+            }.toMutableList()
+        } else {
+            val subDims = dims.subList(1, dims.size)
+
+            return generateSequence {
```

Comment:
Replace with
```
return Array(x) { generateWorld(subDims) }.toMutableList()
```
At this point I have to say. I don't like the data structure `MutableList<Any>`. Lets make our own!